### PR TITLE
Support copy file from node to local host or between nodes.

### LINKED ISSUE #11296: `minikube cp` should support copying from minkube to user's host
currently minikube cp does NOT support doing this.

```
minikube cp minikube:/var/lib/minikube/certs/etcd/peer.crt /tmp
```

current workaround by @tstromberg 
```
minikube ssh "cat /var/lib/minikube/certs/etcd/peer.crt" > /tmp/peer.cert
```

### LINKED ISSUE #11336: Support `cp` between minikube nodes
Related to https://github.com/kubernetes/minikube/issues/11296

It would be nice to have an ability to copy files not only between the host and a node, but also between minikube nodes

`minikube cp node1:/tmp/foo node2:/tmp/bar`

### PR DESCRIPTION
**Pull request by daehyeok**
Fixes #11296 
Fixes #11336 

Support copy file from node to local or node to node.
For the implementation, add new `type ReadableFile interface` to vm_assets and `func (s *SSHRunner) ReadableFile` to SSHRunner. 
 current only SSHRunner used after start, so leave `ReadableFile` as a empty in  ExecRunner and KICRunner. 
 Let me know my implementation not looks good :) 
  thanks

### CODE REVIEW

File: pkg/minikube/command/command_runner.go [hunk 1]
```diff
@@ -77,6 +77,8 @@ type Runner interface {
 
 	// Remove is a convenience method that runs a command to remove a file
 	Remove(assets.CopyableFile) error
+
+	ReadableFile(sourcePath string) (assets.ReadableFile, error)
```
Comment (medyagh) [inline pkg/minikube/command/command_runner.go:81]: comment on this interface, what is this method exactly and is this the best name for it ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: pkg/minikube/command/command_runner.go (hunk 1 above)
Comment (daehyeok) [reply pkg/minikube/command/command_runner.go:81]: I split `assets.CopyableFile` to source(`assets.ReadableFile` interface) and target part in `pkg/minikube/assets/vm_assets.go`.
And this function defined to get source file info/io.Readable interface from remote node. hmm any other suggestion?

File: pkg/minikube/command/ssh_runner.go [hunk 2]
```diff
@@ -49,6 +52,32 @@ type SSHRunner struct {
 	s *ssh.Session
 }
 
+type SSHReadableFile struct {
```
Comment (medyagh) [inline pkg/minikube/command/ssh_runner.go:55]: add comment to all public structs, 
does this need to be public ?

File: pkg/minikube/command/command_runner.go (hunk 1 above)
Comment (medyagh) [reply pkg/minikube/command/command_runner.go:81]: thats fine but add a comment, all public funcs needs to have comment

File: cmd/minikube/cmd/cp_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	"testing"
+)
+
+func TestParsePath(t *testing.T) {
+	var passedCases = []struct {
+		path     string
+		expected remotePath
+	}{
+
+		{"", remotePath{"", ""}},
```
Comment (medyagh) [inline cmd/minikube/cmd/cp_test.go:29]: instead of calling remotePath here, lets add the concrete string here, so the test be simple

**Review by medyagh — CHANGES_REQUESTED** [review]

**Comment by medyagh** [comment]
@daehyeok please pull upstream on this PR

**Comment by daehyeok** [comment]
rebased

**Comment by daehyeok** [comment]
sorry to couldn't care this isuue for while. 

Copy function looks already implemented by other commit.
https://github.com/kubernetes/minikube/commit/e59c41e6f2c9a78be4aff3d0a615ce9f93a49d2f

i will change PR to use new api soon.

**Comment by medyagh** [comment]
@daehyeok sorry for the changes, could u please rebase, we added some API

**Comment by daehyeok** [comment]
Hi i modified my commit and rebased.

on the previous change(e59c41e) `commandRunner` interface have a new api `CopyFrom` and `assets.Copyablefile` include `io.Writer`

but  CopyFrom looks only working for between host and node(not node to node) and `io.Writer` also only used in `ssh_runner.Copyfrom`. I added `BaseCopyableFile` interface and used empty write function to support node to node copy through `ssh_runner`. 
 Maybe we should more clean up those implementation later.

**Comment by medyagh** [comment]
@daehyeok are you still working on this ?

**Pull request merged**